service/history/ndc/history_replicator.go

```diff
@@ -252,7 +252,14 @@ func (r *HistoryReplicatorImpl) ApplyWorkflowState(
 	if err != nil {
 		return err
 	}
-	defer releaseFn(retError)
+	defer func() {
+		if rec := recover(); rec != nil {
+			releaseFn(errPanic)
```

Comment:
errPanic -> rec?

Reply:
releaseFn only care about whether there is an error (to determine whether the clear the cache or not)
the next line will re-throw the panic (so upper layer can log)